R/mail.R

```diff
@@ -247,30 +575,57 @@ download_mod_mbox_per_month <- function(base_url, mailing_list, from_year, to_ye
 #' consistently renamed for clarity.
 #'
 #' @param perceval_path path to perceval binary
-#' @param mbox_path path to mbox archive file (ends in .mbox)
+#' @param mbox_file_path path to mbox archive file (ends in .mbox)
+#' @export
+#' @family parsers
+#' @param perceval_path path to perceval binary
+#' @param mbox_file_path path to mbox archive file (ends in .mbox)
 #' @export
 #' @family parsers
-parse_mbox <- function(perceval_path,mbox_path){
-  # Expand paths (e.g. "~/Desktop" => "/Users/someuser/Desktop")
+parse_mbox <- function(perceval_path, mbox_file_path) {
+  # Expand paths
   perceval_path <- path.expand(perceval_path)
-  mbox_path <- path.expand(mbox_path)
-  # Remove ".mbox"
-  mbox_uri <- stri_replace_last(mbox_path,replacement="",regex=".mbox")
-  # Use percerval to parse mbox_path. --json line is required to be parsed by jsonlite::fromJSON.
-  perceval_output <- system2(perceval_path,
-                             args = c('mbox',mbox_uri,mbox_path,'--json-line'),
-                             stdout = TRUE,
-                             stderr = FALSE)
-  # Parsed JSON output as a data.table.
-  perceval_parsed <- data.table(jsonlite::stream_in(textConnection(perceval_output),verbose=FALSE))
-
-  columns_of_interest <- c("data.Message.ID","data.In.Reply.To","data.Date","data.From","data.To","data.Cc","data.Subject","data.body.plain","data.body")
-  columns_rename <- c("reply_id","in_reply_to_id","reply_datetimetz","reply_from","reply_to","reply_cc","reply_subject","reply_body","reply_body")
+  mbox_file_path <- path.expand(mbox_file_path)
+  mbox_dir <- dirname(mbox_file_path)  # Extract directory path
+  mbox_uri <- mbox_file_path  # URI points to the mbox file
+
+  # Use Perceval to parse the mbox file
+  perceval_output <- tryCatch({
+    system2(perceval_path,
+            args = c('mbox', mbox_uri, mbox_dir, '--json-line'),
+            stdout = TRUE,
+            stderr = TRUE)
+  }, error = function(e) {
+    #print("Error running Perceval:")
+    #print(e$message)
+    stop("Perceval execution failed.")
+  })
+
+  # Filter JSON lines from Perceval output
+  json_lines <- perceval_output[grepl("^\\{", perceval_output)]  # Escape the `{` character
+
+
+  if (length(json_lines) == 0) {
+    stop("No valid JSON lines found in Perceval output. Check the mbox file or Perceval configuration.")
+  }
+
+
+  # Parse JSON output as a data.table
+  perceval_parsed <- tryCatch({
+    data.table(jsonlite::stream_in(textConnection(json_lines), verbose = FALSE))
```

Comment:
@daomcgill I just noticed you are filtering these { outof the json before calling the json parser. Why? This was not in the original.